controllers/util.go

```diff
@@ -86,22 +94,75 @@ func (r *L2ServiceAttachmentReconciler) DefineNetAttachDef(ctx context.Context,
 	// Initiate L2ServiceAttachment Parser
 	l2srvAttParser := l2serviceattachmentparser.NewL2SrvAttParser(s, l2srvObjs, cp, subnets, routesMap, r.Config, r.CniMap, r.IpamMap, log)
 	// Parse the resources and fill the data
-	manifestFolder, err := l2srvAttParser.ParseL2ServiceAttachment(&data)
+	cniManifestFile, ipamManifestFile, err := l2srvAttParser.ParseL2ServiceAttachment(data)
 	if err != nil {
 		log.Error(err, "Error occurred during parsing the L2ServiceAttachment")
 		return nil, err
 	}
 
-	data.Data["NetAttachDefName"] = s.Name
-	data.Data["NetAttachDefNamespace"] = s.Namespace
+	data["NetAttachDefName"] = s.Name
+	data["NetAttachDefNamespace"] = s.Namespace
+	if prefix, ok := data["ResourcePrefix"]; ok {
+		data["NetResourceName"] = prefix.(string) + data["NetObjName"].(string)
+	} else {
+		data["NetResourceName"] = data["NetObjName"].(string)
+	}
+
+	cniFilePath := filepath.Join("manifests", "cni", cniManifestFile)
+	cniConfig, err := getConfig(cniFilePath, data)
 
-	objs, err = render.RenderDir(filepath.Join("manifests", manifestFolder), &data)
 	if err != nil {
 		return nil, err
 	}
+	log.Info("CNI template resolved:", "cni config", cniConfig)
+	data["CNI"] = cniConfig
 
-	ctrl.SetControllerReference(s, objs[0], r.Scheme)
-	return objs[0], nil
+	if ipamManifestFile != "" {
+		ipamFilePath := filepath.Join("manifests", "ipam", ipamManifestFile)
+		ipamConfig, err := getConfig(ipamFilePath, data)
+		if err != nil {
+			return nil, err
+		}
+		log.Info("ipam template resolved:" + cniConfig)
```

Comment:
I guess here you want to write this: 
```suggestion
		log.Info("ipam template resolved:" + ipamConfig)
```

Reply:
Modified